### PR TITLE
Add fpm description and reorganize the README intro

### PR DESCRIPTION
**Pull request by milancurcic**
Sometimes we miss the most fundamental and seemingly obvious thing. Current README does not describe what fpm is or does.

I added one paragraph description to the top of the README, and re-organized a bit what was there previously so that it reads a bit better (IMO).

### CODE REVIEW

**Review by certik — APPROVED** [review]
Looks good, thanks for improving the text.

File: README.md [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-production, we will do our best to stay backwards compatible.
-
-To report a bug report or suggest a feature, please read our
-[contributor guidelines](CONTRIBUTING.md).
+Fortran Package Manager (fpm) is a package manager and build system for Fortran.
+Its key goal is to improve the user experience of Fortran programmers.
+It does so by making it easier to build your Fortran program or library, run the
+executables, tests, and examples, and distribute it as a dependency to other
+Fortran projects.
+Fpm's user interface is modeled after [Rust's Cargo](https://crates.io/),
```
Comment (LKedward) [inline README.md:8]: Is the package registry an appropriate link for Cargo — what about the Cargo book or the Cargo git repo?

**Review by LKedward — APPROVED** [review]
Thanks @milancurcic, this is a nice improvement! Does this close out #382?
Left a small comment about the Cargo link.

File: README.md (hunk 1 above)
Comment (milancurcic) [reply README.md:8]: I thought about that too. I used the top search result for "rust cargo". Cargo book is indeed more appropriate I think

**Comment by milancurcic** [comment]
Yes, I think it should close #382.

**Review by awvwgk — APPROVED** [review]
Reads good. I suggest to mention the quite active discussion board as well.

**Comment by urbanjost** [comment]
Looks good.  Perhaps the phrase "existing packages" should link to the fpm repository.

**Review by urbanjost — APPROVED** [review]

**Review by everythingfunctional — APPROVED** [review]
LGTM

**Comment by awvwgk** [comment]
With five out of five approvals I'll go ahead and merge.

**Pull request merged**